### PR TITLE
Add some more type hints

### PR DESCRIPTION
**Pull request by jan-janssen**

### CODE REVIEW

File: atomistics/calculators/lammps/potential.py [hunk 1]
```diff
@@ -16,15 +17,20 @@ class PotentialAbstract(object):
...
+    ):
         self._potential_df = potential_df
         self._default_df = default_df
         if selected_atoms is not None:
             self._selected_atoms = selected_atoms
         else:
             self._selected_atoms = []
 
-    def find(self, element):
+    def find(self, element: Union[set, list, str]) -> pandas.DataFrame:
```
Comment (liamhuber) [inline atomistics/calculators/lammps/potential.py:33]:
```suggestion
    def find(self, element: Union[set[str], list[str], str]) -> pandas.DataFrame:
```

File: atomistics/calculators/lammps/potential.py [hunk 2]
```diff
@@ -159,7 +164,7 @@ def default(self):
             ]
         return None
 
-    def find_default(self, element):
+    def find_default(self, element: Union[set, list, str]) -> pandas.DataFrame:
```
Comment (liamhuber) [inline atomistics/calculators/lammps/potential.py:167]:
```suggestion
    def find_default(self, element: Union[set[str], list[str], str]) -> pandas.DataFrame:
```

File: atomistics/calculators/hessian.py [hunk 3]
```diff
@@ -101,12 +114,12 @@ def get_pressure_times_volume(stiffness_tensor, structure_equilibrium, structure
...
 @as_task_dict_evaluator
 def evaluate_with_hessian(
-    structure,
-    tasks,
-    structure_equilibrium,
-    force_constants,
-    bulk_modulus=0,
-    shear_modulus=0,
+    structure: Atoms,
+    tasks: dict,
```
Comment (liamhuber) [inline atomistics/calculators/hessian.py:118]: If the type is known already, you can sub-hint `dict[key_type, val_type]`. I only took a quick peek, but I guess here its `dict[str, dict]`.

**Review by liamhuber — APPROVED** [review]
Lgtm, just some suggestions for more specificity.

File: atomistics/calculators/hessian.py (hunk 3 above)
Comment (jan-janssen) [inline atomistics/calculators/hessian.py:118]:
```suggestion
    tasks: dict[str, dict],
```

Comment (jan-janssen) [reply atomistics/calculators/hessian.py]: Done

**Pull request merged**